File: 백준/Silver/10866. 덱/덱.swift
import Foundation

let N = Int(readLine()!)!
var deque:[Int] = []

for _ in 0..<N {
    let cmd = readLine()!.split(separator: " ").map{String($0)}
    switch (cmd[0]) {
        case "push_front":
        var temp = [Int(cmd[1])!]
        for n in deque { temp.append(n) }
        deque = temp

        case "push_back": deque.append(Int(cmd[1])!)

        case "pop_front":
        if (deque.isEmpty) { print(-1) }
        else { print(deque.removeFirst()) }

        case "pop_back":
        if (deque.isEmpty) { print(-1) }
        else { print(deque.removeLast()) }

        case "size": print(deque.count)

        case "empty": print(deque.isEmpty ? 1 : 0)

        case "front":
        if (deque.isEmpty) { print(-1) }
        else { print(deque[0]) }

        case "back":
        if (deque.isEmpty) { print(-1) }
        else { print(deque[deque.count-1]) }

        default: break
    }
}

File: 백준/Silver/10866. 덱/덱.java
import java.util.*;
import java.lang.*;
import java.io.*;

class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int N = Integer.parseInt(br.readLine());
        Deque<Integer> deque = new LinkedList<>();
        
        while(N-- > 0) {
            String s = br.readLine();
            StringTokenizer st = new StringTokenizer(s);
            
            switch(st.nextToken()) {
                case "push_front":
                    deque.offerFirst(Integer.parseInt(st.nextToken()));
                    break;
                case "push_back":
                    deque.offerLast(Integer.parseInt(st.nextToken()));
                    break;
                case "pop_front":
                    System.out.println(deque.isEmpty() ? -1 : deque.pollFirst());
                    break;
                case "pop_back":
                    System.out.println(deque.isEmpty() ? -1 : deque.pollLast());
                    break;
                case "size":
                    System.out.println(deque.size());
                    break;
                case "empty":
                    System.out.println(deque.isEmpty() ? 1 : 0);
                    break;
                case "front":
                    System.out.println(deque.isEmpty() ? -1 : deque.peekFirst());
                    break;
                case "back":
                    System.out.println(deque.isEmpty() ? -1 : deque.peekLast());
                    break;
            }
        }
    }
}